rosidl_typesupport_connext_cpp/resource/msg__type_support.cpp.em

```diff
@@ -165,6 +165,10 @@ publish__@(spec.base_type.type)(
   if (success) {
     @(spec.base_type.pkg_name)::@(subfolder)::dds_::@(spec.base_type.type)_DataWriter * data_writer =
       @(spec.base_type.pkg_name)::@(subfolder)::dds_::@(spec.base_type.type)_DataWriter::narrow(topic_writer);
+    if (!data_writer) {
+      fprintf(stderr, "failed to narrow data writer\n");
+      return false;
```

Comment:
This return leaks the allocated `dds_message`.

Reply:
Oops and thanks! I'll redo it tomorrow.